presto-hive/src/main/java/com/facebook/presto/hive/SubfieldExtractor.java

```diff
@@ -158,4 +165,87 @@ private static boolean hasSubscripts(Optional<Subfield> subfield)
             return Optional.empty();
         }
     }
+
+    public RowExpression toRowExpression(Subfield subfield, Type columnType)
+    {
+        List<Subfield.PathElement> path = subfield.getPath();
+        List<Type> types = new ArrayList<>();
+        types.add(columnType);
+        for (int i = 0; i < path.size(); i++) {
+            Type type = types.get(i);
```

Comment:
seems fragile to be getting this by index from the list as you're creating the list (you're always grabbing the most recently added element, but it seems easy for someone to accidentally mess up later).  I would instead explicitly set it as you create the types (or at the very least explicitly get the last element from the types list).

Reply:
Like so?
> Type type = types.get(types.size()-1);

Reply:
Check my latest version, does it respond well to your concerns?